src/ablation.py

```
@@ -519,6 +519,12 @@ def train_fold(
     else:
         crit = nn.BCEWithLogitsLoss()
 
+    # Auto-enable grad clipping when focal + label smoothing are combined
+    # to prevent gradient explosion from their interaction
+    if loss_type == "focal" and label_smoothing > 0 and grad_clip == 0:
+        grad_clip = 1.0
+        print(f"  Auto-enabled grad_clip=1.0 for focal+smooth stability")
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)